lib/quadquizaminos_web/live/contests_live.ex

```diff
@@ -0,0 +1,132 @@
+defmodule QuadquizaminosWeb.ContestsLive do
+  use Phoenix.LiveView
+  import Phoenix.HTML.Form
+  import Phoenix.LiveView.Helpers
+
+  alias Quadquizaminos.Repo
+
+  alias Quadquizaminos.{Contests, Contest}
+  alias Quadquizaminos.Util
+  alias QuadquizaminosWeb.Router.Helpers, as: Routes
+
+  def mount(_params, session, socket) do
+    :timer.send_interval(1000, self(), :timer)
+    QuadquizaminosWeb.Endpoint.subscribe("timer")
+
+    {:ok,
+     socket |> assign(payloads: [], started_contests: [], contests: Contests.list_contests())}
+  end
+
+  def handle_event("save", %{"key" => "Enter", "value" => contest_name}, socket) do
+    {:noreply, socket |> _create_contest(contest_name)}
+  end
+
+  def handle_event("start", %{"contest" => name}, socket) do
+    socket =
+      if name in Contests.active_contests_names() do
+        Contests.resume_contest(name)
+        socket
+      else
+        _start_contest(socket, name)
+      end
+
+    {:noreply, socket}
+  end
+
+  def handle_event("pause", %{"contest" => name}, socket) do
+    Contests.pause_contest(name)
+    {:noreply, socket}
+  end
+
+  def handle_event("reset", %{"contest" => name}, socket) do
+    Contests.reset_contest(name)
+    {:noreply, socket}
+  end
+
+  def handle_event("stop", %{"contest" => name}, socket) do
+    {:noreply, _end_contest(socket, name)}
+  end
+
+  def handle_info(:timer, socket) do
+    inactive_contest =
```

Comment:
Can we move this logic outside of the callback, maybe into a private function?

Reply:
yes, we can refactor that

Reply:
worked on it